.git_hooks/pre-commit

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env sh
+if ! cargo fmt -- --check ; then
+  printf "\n"
+  printf "\033[0;31mpre-commit hook failed during:\033[0m\n"
+  printf "\033[0;31m\tcargo fmt -- --check\033[0m\n"
```

Comment:
Do these escape characters work on Windows? cc @Kixiron

Reply:
Yah, running them in git bash works. Generally ANSI escapes work everywhere